secret_store/src/key_server_cluster/math.rs

```diff
@@ -681,12 +875,63 @@ pub mod tests {
 		}
 	}
 
+	#[test]
+	fn full_ecdsa_signature_math_session() {
+		let test_cases = [(2, 5), (2, 6), (3, 11), (4, 11)];
+		for &(t, n) in &test_cases {
+			// values that can be hardcoded
+			let joint_secret: Secret = Random.generate().unwrap().secret().clone();
+			let joint_nonce: Secret = Random.generate().unwrap().secret().clone();
+			let message_hash: H256 = H256::random();
+
+			// convert message hash to EC scalar
+			let message_hash_scalar = to_scalar(message_hash.clone()).unwrap();
+
+			// generate secret key shares
+			let artifacts = run_key_generation(t, n, None, Some(joint_secret));
+
+			// generate nonce shares
+			let nonce_artifacts = run_key_generation(t, n, Some(artifacts.id_numbers.clone()), Some(joint_nonce));
+
+			// compute nonce public
+			// x coordinate (mapped to EC field) of this public is the r-portion of signature
+			let nonce_public_shares: Vec<_> = (0..n).map(|i| compute_public_share(&nonce_artifacts.polynoms1[i][0]).unwrap()).collect();
+			let nonce_public = compute_joint_public(nonce_public_shares.iter()).unwrap();
+			let signature_r = compute_ecdsa_r(&nonce_public).unwrap();
+
+			// compute shares of inv(nonce) so that both nonce && inv(nonce) are still unknown to all nodes
+			let nonce_inv_shares = run_reciprocal_protocol(t, &nonce_artifacts);
+
+			// compute multiplication of secret-shares * inv-nonce-shares
+			let mul_shares = run_multiplication_protocol(t, &artifacts.secret_shares, &nonce_inv_shares);
+
+			// compute shares for s portion of signature: nonce_inv * (message_hash + secret * signature_r)
+			// every node broadcasts this share
```

Comment:
if each node's participation (or at least 2t + 1) is required to create a share here, then can it really be considered a t-of-n scheme?

Reply:
Replaced with `2*t+1` (it is actually `compute_ecdsa_s` who take care of this, but it might be confusing to see n in this test)